packit_service/worker/handlers/distgit.py

```diff
@@ -806,6 +829,14 @@ def run(self) -> TaskResults:
                 )
                 continue
 
+            if self.is_already_triggered(branch):
```

Comment:
I think we should skip this if `self.job_config.scratch` it set (scratch builds can have the same NVR as existing production builds and `KojiHelper.get_build_info()` ignores scratch builds).